compat/src/render.js

```diff
@@ -0,0 +1,142 @@
+import {
+	render as preactRender,
+	options,
+	toChildArray,
+	Component
+} from 'preact';
+import { applyEventNormalization } from './events';
+
+const CAMEL_PROPS = /^(?:accent|alignment|arabic|baseline|cap|clip|color|fill|flood|font|glyph|horiz|marker|overline|paint|stop|strikethrough|stroke|text|underline|unicode|units|v|vector|vert|word|writing|x)[A-Z]/;
+
+// Some libraries like `react-virtualized` explicitly check for this.
+Component.prototype.isReactComponent = {};
+
+/* istanbul ignore next */
+export const REACT_ELEMENT_TYPE =
+	(typeof Symbol !== 'undefined' &&
+		Symbol.for &&
+		Symbol.for('react.element')) ||
+	0xeac7;
+
+/**
+ * Proxy render() since React returns a Component reference.
+ * @param {import('./internal').VNode} vnode VNode tree to render
+ * @param {import('./internal').PreactElement} parent DOM node to render vnode tree into
+ * @param {() => void} [callback] Optional callback that will be called after rendering
+ * @returns {import('./internal').Component | null} The root component reference or null
+ */
+export function render(vnode, parent, callback) {
+	// React destroys any existing DOM nodes, see #1727
+	// ...but only on the first render, see #1828
+	if (parent._children == null) {
+		while (parent.firstChild) {
+			parent.removeChild(parent.firstChild);
```

Comment:
Most of the byte savings come from removing the call to `preact/src/util.removeNode` here and just calling `removeChild` directly since we are looping through the parent element anyway: b1443c5